system/include/compat/xmmintrin.h

```diff
@@ -605,9 +607,11 @@ static __inline__ int __attribute__((__always_inline__, __nodebug__, DIAGNOSE_SL
 
 static __inline__ int __attribute__((__always_inline__, __nodebug__, DIAGNOSE_SLOW)) _mm_cvttss_si32(__m128 __a)
 {
-  int x = lrint(((__f32x4)__a)[0]);
-  if (x != 0 || fabsf(((__f32x4)__a)[0]) < 2.f)
-    return (int)((__f32x4)__a)[0];
+  float e = ((__f32x4)__a)[0];
+  if (isnanf(e) || e > INT_MAX || e < INT_MIN) return (int)0x80000000;
+  int x = lrint(e);
+  if ((x != 0 || fabsf(e) < 2.f))
+    return (int)e;
```

Comment:
if we used `__builtin_wasm_trunc_s_i32_f32` instead of just the C cast here, we could avoid having LLVM insert checks redundant to these in the backend.

Reply:
Aren't those checks going away though?

Reply:
Done!